tests/robustness/model/replay.go

```diff
@@ -99,13 +99,26 @@ func (r *EtcdReplay) next() (request EtcdRequest, revision int64, index int) {
 
 type WatchEvent struct {
 	Event
-	Revision int64
+	Revision    int64
+	CreateEvent bool
+}
+
+func (we WatchEvent) Equals(event WatchEvent) bool {
+	doPrevValsMatch := we.PrevValue == event.PrevValue ||
+		we.PrevValue != nil && event.PrevValue != nil && *we.PrevValue == *event.PrevValue
+
+	return we.Revision == event.Revision &&
+		we.Type == event.Type &&
+		we.Key == event.Key &&
+		we.Value == event.Value &&
+		doPrevValsMatch
 }
 
 type Event struct {
-	Type  OperationType
-	Key   string
-	Value ValueOrHash
+	Type      OperationType
+	Key       string
+	Value     ValueOrHash
+	PrevValue *ValueRevision
```

Comment:
I would expect that `PrevValue` should be on `WatchEvent`. What was the reason to put it here?

Reply:
Possibly we will need to split off another Event type.
* `Event` that has `Type`, `Key`, `Value`
* `PersistedEvent` that has `Event`, `Revision` and `IsCreate`.
* `WatchEvent` that has `PersistedEvent` and `PrevValue`

Reply:
Its probably better to split this off into another PR and then rebase that into this once merged. I'll send one in.